### PR TITLE
Added test for Try.

### PR DESCRIPTION
**Pull request by aballano**
Fixed a small bug in the way.

Note: I did not use the `forAll` method because I don't see any benefit since we don't operate with the value itself but with the types. Also because adding this method basically makes very difficult to diagnose any test failure.

### CODE REVIEW

File: katz/src/main/kotlin/katz/data/Try.kt [hunk 1]
```diff
@@ -80,18 +80,21 @@ sealed class Try<out A> {
...
      */
-    class Success<out A>(val value: A) : Try<A>() {
-        override val isFailure: Boolean = true
-        override val isSuccess: Boolean = false
+    data class Success<out A>(val value: A) : Try<A>() {
+        override val isFailure: Boolean = false
+        override val isSuccess: Boolean = true
     }
+
+    data class PredicateException(override val message: String) : Throwable(message)
```
Comment (raulraja) [inline katz/src/main/kotlin/katz/data/Try.kt]: let's make this a sealed hierarchy so consumers matching over the exceptions get full compiler support to deal with all potential datatypes.

Comment (aballano) [inline katz/src/main/kotlin/katz/data/Try.kt]: Change done!

File: katz/src/main/kotlin/katz/data/Try.kt [hunk 2]
```diff
@@ -74,26 +78,35 @@ sealed class Try<out A> {
...
 
     /**
      * The `Failure` type represents a computation that result in an exception.
      */
-    class Failure<out A>(val exception: Throwable) : Try<A>() {
-        override val isFailure: Boolean = false
-        override val isSuccess: Boolean = true
+    data class Failure<out A>(val exception: Throwable) : Try<A>() {
+        override val isFailure: Boolean = true
+        override val isSuccess: Boolean = false
```
Comment (pakoito) [inline katz/src/main/kotlin/katz/data/Try.kt]: These booleans look like a possible scape hatch we don't really need. They could either be retrieved by a fold, or simply using when.

Comment (aballano) [inline katz/src/main/kotlin/katz/data/Try.kt]: Fine for me, I think we put them there as a mirror of the Scala implementation, but I agree that I don't see the usage having other methods. Let's collect some more opinions before removing them, tho.

@FineCinnamon/katzmembers Please vote with :+1: or :-1: to remove those fields.

**Pull request merged**